### ISSUE TITLE
Some questions about some codes in the implementation

### ISSUE THREAD

**Issue by HelenMao** [1Konny/FactorVAE#10]
Hi, thanks for your great implementation of Factor-VAE.
I want to introduce the total correlation loss in my work so that I read your implementation carefully.
However, there are some codes I do not understand.
In solver.py line107
``
 vae_tc_loss = (D_z[:, :1] - D_z[:, 1:]).mean()
``
why you use this to compute the tc_loss for VAE
and line 119
``
D_tc_loss = 0.5*(F.cross_entropy(D_z, zeros) + F.cross_entropy(D_z_pperm, ones))
``
when optimizing the discriminator, why the label of D_z is zeros and D_z_pperm is ones.
Looking forward to your response, thanks!

**Comment by speedcell4** [comment]
@HelenMao did you figure that out? I get focused on that too

**Comment by HelenMao** [comment]
Yes, the paper gives some implementation details.
E.g. Figure1 said that the label of D_z is zeros and D_z_pperm is ones.

**Comment by speedcell4** [comment]
Thank you so much.
By the way, did you implement the metric mentioned in Figure 2?

**Comment by HelenMao** [comment]
No, I didn't.

**Comment by Natsu6767** [comment]
> Yes, the paper gives some implementation details.
> E.g. Figure1 said that the label of D_z is zeros and D_z_pperm is ones.

However, I believe that according to **Algorithm 2** in the paper, the label for _D_z_ should be ones and that for _D_z_pperm_ should be zeros.

**Comment by a514514772** [comment]
I agree with @Natsu6767. There seems to be a typo.

Can someone please tell me why the following code is implemented in this way?
`vae_tc_loss = (D_z[:, :1] - D_z[:, 1:]).mean()`

**Comment by HelenMao** [comment]
In fact, the codes use F.cross_entropy this function. 
The second input is target which means the target class.
The output of the discriminator is two classes.
zero indicates the first class, and one indicates the second class.
